core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4903,7 +4903,7 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
    * Prefix form of `ZIO#uninterruptible`.
    */
   def uninterruptible[R, E, A](zio: => ZIO[R, E, A])(implicit trace: Trace): ZIO[R, E, A] =
-    ZIO.suspendSucceed(zio.uninterruptible)
+    ZIO.suspendSucceed(zio).uninterruptible
```

Comment:
This really didn't seem right to me. I think we should be making the outer effect uninterruptible, not the inner one. Otherwise, when the effect is run, if the outer one is not uninterruptible we will get an interruption

Reply:
I agree, people might have imperative code hidden in the evaluation of that ZIO.